### PR TITLE
Parameterize Simulator and Cluster in test_raftstore-v2

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What is changed and how it works?

Issue Number: ref #14401

What's Changed:

Parameterize Simulator and Cluster in test_raftstore-v2.

```commit-message
Simulator and Cluster in test_raftstore-v2
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/test_raftstore-v2/Cargo.toml [hunk 1]
```diff
@@ -66,3 +66,8 @@ tikv_util = { workspace = true }
 tokio = { version = "1.5", features = ["rt-multi-thread"] }
 tokio-timer = { git = "https://github.com/tikv/tokio", branch = "tokio-timer-hotfix" }
 txn_types = { workspace = true }
+
+[dependencies.rocksdb]
```
Comment (SpadeA-Tang) [inline components/test_raftstore-v2/Cargo.toml:70]: why need this?

Comment (SpadeA-Tang) [inline components/test_raftstore-v2/Cargo.toml:70]: why need this

File: components/test_raftstore-v2/src/lib.rs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 #![feature(type_alias_impl_trait)]
 #![feature(return_position_impl_trait_in_trait)]
 #![feature(let_chains)]
+#![feature(trait_alias)]
```
Comment (SpadeA-Tang) [inline components/test_raftstore-v2/src/lib.rs:6]: not used

File: components/test_raftstore-v2/Cargo.toml (hunk 1 above)
Comment (SpadeA-Tang) [inline components/test_raftstore-v2/Cargo.toml:70]: why need this

**Review by tonyxuqqi — APPROVED** [review]

File: components/test_raftstore/src/cluster.rs [hunk 3]
```diff
@@ -1947,7 +1947,9 @@ impl<T: Simulator> Drop for Cluster<T> {
     }
 }
 
-pub trait RawEngine: Peekable<DbVector = RocksDbVector> + SyncMutable {
+pub trait RawEngine<EK: engine_traits::KvEngine>:
```
Comment (tabokie) [inline components/test_raftstore/src/cluster.rs:1950]: Why is this necessary?

**Review by tabokie — APPROVED** [review]
